### PR TITLE
Add relative coordinates component to Wiremod

### PR DESCRIPTION
**Pull request by Dejaku51**
## About The Pull Request
Adds relative coords component which requires line of sight with target to work
Changed can_see default value of length var to match world.view instead of just 5

## Why It's Good For The Game
Helps people make funnier circuits

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

image

image

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl:
add: Added relative coords components which lets you get relative coordinates to target from circuit POV (requires line of sight)
code: Changed can_see default value of length var to 7
/:cl:

### CODE REVIEW

**Review by itsmeow — APPROVED** [review]
Hell yeah

File: code/modules/wiremod/components/atom/relative_coords.dm [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	calculated = add_output_port("Calculated", PORT_TYPE_SIGNAL)
+	failed = add_output_port("Failed", PORT_TYPE_SIGNAL)
+
+
+
+/obj/item/circuit_component/relative_coords/input_received(datum/port/input/port, list/return_values)
+	if(COMPONENT_TRIGGERED_BY(calculate, port))
+		var/atom/target_atom = get_turf(target.value)
+		var/atom/source = get_turf(parent)
+		if(can_see(parent, target_atom)) // requires line of sight
```
Comment (itsmeow) [inline code/modules/wiremod/components/atom/relative_coords.dm:38]: this only has a range of 5 by default. Should we change it to `world.view`? (7)

Comment (Dejaku51) [reply code/modules/wiremod/components/atom/relative_coords.dm:38]: changed it so can_see uses world.view by default instead of 5

Comment (itsmeow) [reply code/modules/wiremod/components/atom/relative_coords.dm:38]: btw world.view is not always an integer

Comment (Dejaku51) [reply code/modules/wiremod/components/atom/relative_coords.dm:38]: so want me to just make it 7?

Comment (Dejaku51) [reply code/modules/wiremod/components/atom/relative_coords.dm:38]: 7 it is

**Pull request merged**